### ISSUE TITLE
Emscripten SDL_GetDisplayBounds returns incorrect display height in itch.io

### ISSUE THREAD

**Issue by martinstarkov** [libsdl-org/SDL#11040]
**Context:**
- Building with emcc and uploading the zipped html/js files to itch.io, which allows embedded HTML game files to be played through their website.
- Using a Windows 10 laptop (single display) with 1920x1080 resolution. 

**Issue:**
SDL2 correctly identifies the 1920x1080 resolution for my display when running the application as an exe and local hosting the emscripten HTML files using emrun. However, when uploading the HTML files to itch, the same SDL_GetDisplayBounds call returns 1920x1040. Below is a fullscreen image of my entire display when running inside a 900x900 canvas (itch allows canvas size to be specified when uploading the zip). The grey area is the 900x900 canvas, and the bordered pink area is the 800x800 viewport (custom OpenGL renderer). 

![image](https://github.com/user-attachments/assets/09cab80c-30de-4e2a-a45a-c29287b18c17)

As a result, when I fullscreen the canvas using ```SDL_SetWindowFullscreen(window, SDL_WINDOW_FULLSCREEN_DESKTOP)```, the window size is only set to 1920x1040 which messes with how I position the viewport in the top left of the screen:

![image](https://github.com/user-attachments/assets/f4e36ddb-efbc-4b28-8338-3fa43c3b0882)

However as seen in the image, the canvas size and CSS style retrieved by emscripten are 1920x1080. When attempting to set the window size manually using SDL_SetWindowSize, nothing happens. I suspect this has something to do with how inside SDL_SetWindowSize the code has:
```c++
if (window->flags & SDL_WINDOW_FULLSCREEN) {
        if (FULLSCREEN_VISIBLE(window) && (window->flags & SDL_WINDOW_FULLSCREEN_DESKTOP) != SDL_WINDOW_FULLSCREEN_DESKTOP) {
            window->last_fullscreen_flags = 0;
            SDL_UpdateFullscreenMode(window, SDL_TRUE);
        }
    } else {
    ...
   ```
Which essentially skips updating the window size for SDL_WINDOW_FULLSCREEN_DESKTOP.

I briefly looked into how SDL uses the emscripten fullscreen resize callback to manage the size of SDL_WINDOW_FULLSCREEN_DESKTOP. The issue is that the transition from windowed to desktop fullscreen goes through multiple resizes (800, 800) -> (1920, 925) -> (1920, 1080), which triggers the resize callback multiple times, however, the final canvas size is reached only after the fullscreen change has occurred, which omits the call to the fullscreen resize callback. This may be entirely unrelated to the SDL_GetDisplayBounds issue, but highlights how bandaiding the issue is not possible with the current implementation of SDL_SetWindowSize / SDL_WINDOW_FULLSCREEN_DESKTOP.

[Here is a link to the canvas page](https://bicyclemice.itch.io/protegon)

I inspected it with dev tools and it looks like there is no padding/margins. Any idea why SDL_GetDisplayBounds would think the display resolution is 1920x1040 only on itch.io?

Tested on SDL 2.30.0 and SDL 2.30.8 (not sure which version emscripten's --use-port=sdl2 uses).

**Comment by martinstarkov** [comment]
I've uploaded a minimal example of the bug here:
https://github.com/martinstarkov/emscripten-sdl-bug
And the itch page for the above code here:
https://bicyclemice.itch.io/emscripten-bug

**Comment by martinstarkov** [comment]
I believe I've identified the cause of the issue. It seems as though within itch.io there is a div that has ```padding-bottom: 40px``` set. Removing this before running the game removes the issue.
![image](https://github.com/user-attachments/assets/e6a324f3-ee3b-4195-930e-84927f009149)
I'm still wondering why a padded div would modify the output of SDL_GetDisplayBounds.

**Comment by martinstarkov** [comment]
It seems like SDL_GetDisplayBounds is locked as whatever value it retrieved when the page was loaded. So for instance if I resize the browser down and hit "run game" it will show a display bounds of e.g. 1500x900, but if I then fullscreen the browser, the display bounds remains as 1500x900.

**Comment by martinstarkov** [comment]
Alright so the emscripten video display is added based on the output of ```emscripten_get_screen_size``` by Emscripten_VideoInit.
```c++
// from SDL_emscriptenvideo.c
int Emscripten_VideoInit(_THIS)
{
    SDL_DisplayMode mode;

    /* Use a fake 32-bpp desktop mode */
    mode.format = SDL_PIXELFORMAT_RGB888;
    emscripten_get_screen_size(&mode.w, &mode.h);

    mode.refresh_rate = 0;
    mode.driverdata = NULL;
    if (SDL_AddBasicVideoDisplay(&mode) < 0) {
        return -1;
    }

    SDL_AddDisplayMode(&_this->displays[0], &mode);

    Emscripten_InitMouse();

    /* We're done! */
    return 0;
}

```
However, Emscripten_SetDisplayMode is left blank:
```c++
static int Emscripten_SetDisplayMode(_THIS, SDL_VideoDisplay *display, SDL_DisplayMode *mode)
{
    /* can't do this */
    return 0;
}
```
From what I can tell,  ```emscripten_get_screen_size``` returns the size of the browser instead of the screen size when in itch.

**Comment by martinstarkov** [comment]
Indeed, it seems that when calling ```emscripten_get_screen_size``` from inside an itch canvas, it retrieves the size of the browser window instead of the display size. And when combining this with the fact that the SDL_SetWindowSize implementation does not allow overriding desktop fullscreen size, it becomes impossible to size the window up manually
![image](https://github.com/user-attachments/assets/a4d23104-baed-4695-ac28-8287568ee627)

**Comment by martinstarkov** [comment]
[Back in 2021](https://github.com/libsdl-org/SDL/commit/d8baad7e483ff04c0bbd7fa8d6d49dce243376df), Emscripten_VideoInit still used:

```c++
  mode.w = EM_ASM_INT_V({
      return screen.width;
  });
  mode.h = EM_ASM_INT_V({
      return screen.height;
  });
```
instead of ```emscripten_get_screen_size```. I tested the output of this and it produced the same result:
![image](https://github.com/user-attachments/assets/4ef971c2-c482-4267-8bd2-8e86c7e094be)

**Comment by martinstarkov** [comment]
Tested on Chrome and it works fine:
![image](https://github.com/user-attachments/assets/c392cdf9-9851-4b7b-912d-45ee13291e8f)
It seems to have been a Brave browser issue at some point: https://community.brave.com/t/screen-width-and-screen-height-are-wrong/446476
Looks like they fixed it, which explains why the locally run canvas works fine, however when uploading to itch Brave's screen.width no longer reports the display width, but instead the browser width.

Hopefully, if someone runs into this extremely niche behavior the keywords in this issue will save them some headache.

In other words, nothing to do with SDL2.
